File: android/app/src/main/java/com/nexa/social/data/api/AuthApi.kt
package com.nexa.social.data.api

import com.nexa.social.data.models.ApiResponse
import com.nexa.social.data.models.FcmTokenRequest
import com.nexa.social.data.models.ForgotPasswordRequest
import com.nexa.social.data.models.LoginRequest
import com.nexa.social.data.models.LoginResponse
import com.nexa.social.data.models.RefreshTokenRequest
import com.nexa.social.data.models.RefreshTokenResponse
import com.nexa.social.data.models.RegisterRequest
import com.nexa.social.data.models.ResetPasswordRequest
import com.nexa.social.data.models.VerifyEmailRequest
import retrofit2.Response
import retrofit2.http.Body
import retrofit2.http.DELETE
import retrofit2.http.POST

interface AuthApi {
    @POST("auth/register")
    suspend fun register(@Body request: RegisterRequest): Response<ApiResponse<LoginResponse>>

    @POST("auth/login")
    suspend fun login(@Body request: LoginRequest): Response<ApiResponse<LoginResponse>>

    @POST("auth/refresh")
    suspend fun refresh(@Body request: RefreshTokenRequest): Response<ApiResponse<RefreshTokenResponse>>

    @POST("auth/logout")
    suspend fun logout(@Body request: Map<String, String>? = null): Response<ApiResponse<Map<String, Any>>>

    @POST("auth/forgot-password")
    suspend fun forgotPassword(@Body request: ForgotPasswordRequest): Response<ApiResponse<Map<String, Any>>>

    @POST("auth/reset-password")
    suspend fun resetPassword(@Body request: ResetPasswordRequest): Response<ApiResponse<Map<String, Any>>>

    @POST("auth/verify-email")
    suspend fun verifyEmail(@Body request: VerifyEmailRequest): Response<ApiResponse<Map<String, Any>>>

    @POST("auth/resend-verification")
    suspend fun resendVerification(@Body request: Map<String, String>): Response<ApiResponse<Map<String, Any>>>

    @POST("notifications/register")
    suspend fun registerFcmToken(@Body request: FcmTokenRequest): Response<ApiResponse<Map<String, Any>>>

    @DELETE("notifications/fcm-token")
    suspend fun revokeFcmToken(@Body request: FcmTokenRequest): Response<ApiResponse<Map<String, Any>>>
}

File: client/src/api/auth.api.ts
import { api, setAccessToken, clearAuthSession } from './client.js';
import { User, ApiResponse } from '../types/index.js';

export const authApi = {
  async register(data: {
    username: string;
    email: string;
    password: string;
    displayName: string;
    bio?: string;
  }): Promise<{ user: User; accessToken: string; refreshToken?: string }> {
    const res = await api.post<ApiResponse<{ user: User; accessToken: string; refreshToken?: string }>>(
      '/auth/register',
      data
    );
    setAccessToken(res.data.data.accessToken);
    return res.data.data;
  },

  async login(data: {
    emailOrUsername: string;
    password: string;
  }): Promise<{ user: User; accessToken: string; refreshToken?: string }> {
    const res = await api.post<ApiResponse<{ user: User; accessToken: string; refreshToken?: string }>>(
      '/auth/login',
      data
    );
    setAccessToken(res.data.data.accessToken);
    return res.data.data;
  },

  async refresh(): Promise<string> {
    const res = await api.post<ApiResponse<{ accessToken: string; refreshToken?: string }>>('/auth/refresh', {});
    setAccessToken(res.data.data.accessToken);
    return res.data.data.accessToken;
  },

  async logout(): Promise<void> {
    try {
      await api.post('/auth/logout', {});
    } finally {
      clearAuthSession();
    }
  },

  async me(): Promise<User> {
    const res = await api.get<ApiResponse<User>>('/auth/me');
    return res.data.data;
  },

  async forgotPassword(email: string): Promise<{ message: string }> {
    const res = await api.post<ApiResponse<{ message: string }>>('/auth/forgot-password', { email });
    return res.data.data;
  },

  async resetPassword(data: { token: string; newPassword: string }): Promise<{ message: string }> {
    const res = await api.post<ApiResponse<{ message: string }>>('/auth/reset-password', data);
    return res.data.data;
  },

  async verifyEmail(token: string): Promise<{ success: boolean; message: string }> {
    const res = await api.post<ApiResponse<{ success: boolean; message: string }>>('/auth/verify-email', { token });
    return res.data.data;
  },

  async resendVerification(email?: string): Promise<{ message: string }> {
    const res = await api.post<ApiResponse<{ message: string }>>('/auth/resend-verification', { email });
    return res.data.data;
  }
};
